core/polyaxon/connections/gcp/gcs.py

```diff
@@ -339,29 +339,17 @@ def delete(self, key, bucket_name=None, workers=0):
         if not bucket_name:
             bucket_name, key = self.parse_gcs_url(key)
 
-        results = self.list(bucket_name=bucket_name, key=key, delimiter="/")
-        if not any([results["prefixes"], results["blobs"]]):
-            self.delete_file(key=key, bucket_name=bucket_name)
-
-        # Delete directories
-        for prefix in sorted(results["prefixes"]):
-            prefix = os.path.join(key, prefix)
-            # Download files under
-            self.delete(key=prefix, bucket_name=bucket_name)
-
+        blobs = self.connection.list_blobs(bucket_name, prefix=key)
         pool, future_results = self.init_pool(workers)
 
         # Delete files
-        for file_key in results["blobs"]:
-            file_key = file_key[0]
-            file_key = os.path.join(key, file_key)
+        for blob in blobs:
             future_results = self.submit_pool(
                 workers=workers,
                 pool=pool,
                 future_results=future_results,
-                fn=self.delete_file,
-                key=file_key,
-                bucket_name=bucket_name,
+                fn=_delete_blob,
+                blob=blob,
```

Comment:
I am not very familiar with the thread pool. Is is problematic to pass around the entire `blob` class?
If so this could be changed to `bucket.delete_blob(blob_name)`.